File: Hard/2025-10-22-3347-Maximum-Frequency-of-an-Element-After-Performing-Operations-II/solution.java
import java.util.Arrays;

public class Solution {
    public int maxFrequency(int[] nums, int k, int numOperations) {
        Arrays.sort(nums);
        int left = 0, right = 0, n = nums.length;
        long total = 0;
        int maxFreq = 0;

        while (right < n) {
            total += nums[right];
            while (nums[right] * (right - left + 1) - total > k * numOperations) {
                total -= nums[left];
                left++;
            }
            maxFreq = Math.max(maxFreq, right - left + 1);
            right++;
        }

        return maxFreq;
    }
}

File: Hard/2025-10-22-3347-Maximum-Frequency-of-an-Element-After-Performing-Operations-II/solution.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int, numOperations: int) -> int:
        nums.sort()
        left = 0
        total = 0
        max_freq = 0
        
        for right in range(len(nums)):
            total += nums[right]
            while nums[right] * (right - left + 1) - total > k * numOperations:
                total -= nums[left]
                left += 1
            max_freq = max(max_freq, right - left + 1)
        
        return max_freq